examples/pbr/main.js

```diff
@@ -0,0 +1,1411 @@
+                        osg.Matrix.makeTranslate( x, -y * 1.2, 0, sample.getMatrix() );
+
+                        roughnessTexture = this.createTextureFromColor( roughness, false );
+
+                        this.setMaterial( sample.getOrCreateStateSet(), albedo, roughnessTexture, metalTexture );
+
+                        group.addChild( sample );
+                    }
+                }
+
+            }
+
+            return group;
+        },
+
+        createSampleModels: function () {
+
+            var nb = 8;
+            var offset = 8 * 20;
+
+            var group = new osg.Node();
+
+            var stateSet;
+            var config;
+
+            var rowRoughness = this.createRowModelsRoughness( nb, offset );
+            stateSet = rowRoughness.getOrCreateStateSet();
+            config = {
+                stateSet: stateSet,
+                config: {
+                    noTangent: true
+                }
+            };
+            this._shaders.push( config );
+            group.addChild( rowRoughness );
+            osg.Matrix.makeTranslate( 0, 0, 0, rowRoughness.getMatrix() );
+
+            if ( false )
+                return group;
+
+            var rowMetalic = this.createRowModelsMetalic( nb, offset );
+            stateSet = rowMetalic.getOrCreateStateSet();
+            config = {
+                stateSet: stateSet,
+                config: {
+                    noTangent: true
+                }
+            };
+            this._shaders.push( config );
+            group.addChild( rowMetalic );
+            osg.Matrix.makeTranslate( 0, 40, 0, rowMetalic.getMatrix() );
+
+            var rowSpecular = this.createRowModelsSpecularMetal( nb, offset );
+            stateSet = rowSpecular.getOrCreateStateSet();
+            config = {
+                stateSet: stateSet,
+                config: {
+                    specularMap: true,
+                    noTangent: true
+                }
+            };
+            this._shaders.push( config );
+            group.addChild( rowSpecular );
+            osg.Matrix.makeTranslate( 0, 80, 0, rowSpecular.getMatrix() );
+
+
+            this.updateShaderPBR();
+
+            group.getOrCreateStateSet().setAttributeAndModes( new osg.CullFace() );
+            return group;
+        },
+
+
+        createSampleScene: function () {
+
+            var group = new osg.Node();
+            this._mainSceneNode = group;
+            // add environment geometry
+            var environmentGeometry = this.createEnvironmentNode();
+            group.addChild( environmentGeometry );
+
+            this._environmentStateSet = environmentGeometry.getOrCreateStateSet();
+
+            group.addChild( this.createSampleModels() );
+
+            // add node that contains model loaded
+            group.addChild( this._proxyRealModel );
+
+            return group;
+        },
+
+        createShaderPanorama: function ( defines ) {
```

Comment:
deriving from example template would allow all the "createShader*" funcs to be replaced with example::createShader https://github.com/cedricpinson/osgjs/blob/develop/examples/templates/Example.js#L159